crates/signer/src/signer.rs

```diff
@@ -0,0 +1,116 @@
+use alloy_signer_local::PrivateKeySigner;
+use eth_keystore::decrypt_key;
+use std::path::Path;
+use thiserror::Error;
+
+#[derive(Debug)]
+/// Represents the input params to create a signer
+pub enum Config {
+    /// Hexadecimal private key
+    PrivateKey(String),
+    /// Keystore path and password
+    Keystore(String, String),
+    /// Web3 endpoint and address
+    Web3(String, String),
+}
+
+#[derive(Error, Debug)]
+#[allow(missing_docs)]
+pub enum SignerError {
+    #[error("invalid private key")]
+    InvalidPrivateKey,
+    #[error("invalid keystore password")]
+    InvalidPassword,
+}
+
+impl Config {
+    /// Creates a signer from given config.
+    pub fn signer_from_config(c: Config) -> Result<PrivateKeySigner, SignerError> {
+        // TODO: check chain id to select signer
```

Comment:
Check if an equivalent of [LatestSignerForChainID](https://github.com/ethereum/go-ethereum/blob/master/core/types/transaction_signing.go#L91) exists for rust